account_invoice_change_currency/models/account_change_currency.py

```diff
@@ -0,0 +1,85 @@
+# -*- encoding: utf-8 -*-
+from openerp import fields, models, api, _
+from openerp.exceptions import ValidationError
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    old_currency_id = fields.Many2one('res.currency')
+    custom_rate = fields.Float(compute='_compute_custome_rate_currency',
+                               inverse='_inverse_custom_rate_currency',
+                               required=True, help="Set new currency rate to "
+                               "apply on the invoice\n.This rate will be taken"
+                               " in order to convert amounts between the "
+                               "currency on the invoice and MX currency",
+                               digits=(12, 6), store=True)
+
+    @api.model
+    def create(self, values):
+        values['old_currency_id'] = values['currency_id']
+        return  super(AccountInvoice, self).create(values)
```

Comment:
I do not line to write a create for this, in the onchange check if is set, if dont then is the company one or something like that.